tasks/main.yml

```diff
@@ -8,6 +8,23 @@
 - name: Install Docker.
   package: name={{ docker_package }} state={{ docker_package_state }}
 
+- name: Ensure Docker directory exists.
+  file:
+    path: /etc/docker
+    state: directory
+    owner: root
+    group: root
+    mode: 0700
+
+- name: Ensure Docker configuration is present.
+  template:
+    src: etc/docker/daemon.json.j2
```

Comment:
If I'm going to include any template with this role, I'd like to do it in as flexible a manner as possible. I do like the idea of having at least a few sane defaults (I have often had to switch on Google DNS for container resolution to work...), but it should be easy for someone to completely override the template, and/or possibly blocks inside the template.

For this example taken to the extreme, check out my Nginx role: https://github.com/geerlingguy/ansible-role-nginx#overriding-configuration-templates

For a simpler example, check out my Varnish role: https://github.com/geerlingguy/ansible-role-varnish#role-variables (the `varnish_default_vcl_template_path`)